### PR TITLE
Fix the Hasse diagram widget to support facts

### PR DESCRIPTION
**Pull request by nomeata**

### CODE REVIEW

**Comment by nomeata** [comment]
@0art0, I broke your widget in #130, sorry for that. This is a (blind) attempt to fix it

**Comment by nomeata** [comment]
Although I don’t know how well that code will hande the millions of edges we now have :-)

File: widget/src/HasseDiagram.tsx [hunk 1]
```diff
@@ -34,10 +39,14 @@ const Diagram: React.FC<DiagramProps> = ({ implications, nonimplications }) => {
...
 
       // Add nodes and edges for non-implications
-      nonimplications.forEach(({ lhs, rhs }) => {
-        addNodeIfNotExists(lhs, lhs);
-        addNodeIfNotExists(rhs, rhs);
-        edges.add({ id: `nonimplication-${lhs}-${rhs}`, from: lhs, to: rhs, arrows: 'to', color: '#ff9999' }); // Non-implication edge color
+      facts.forEach(({ satisfied, refuted }) => {
+        satisfied.forEach(lhs => {
+          refuted.forEach(rhs => {
+            addNodeIfNotExists(lhs, lhs);
```
Comment (eric-wieser) [inline widget/src/HasseDiagram.tsx:45]:
```suggestion
          addNodeIfNotExists(lhs, lhs);
          refuted.forEach(rhs => {
```

**Comment by eric-wieser** [comment]
> Also, could you add a README with instructions on how to build the widget? That would help the TS-illiterate like me a lot.

I think the instructions are:
```
$ curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.1/install.sh | bash
$ nvm install node
$ cd widget
widget$ npm run build
```

**Comment by nomeata** [comment]
Ok, have to find out how to translate that to nixos world :-) (If somenoe wants to take over this PR, happyily)

**Comment by 0art0** [comment]
I'm sorry for joining in late on this thread. Eric's instructions for building the JavaScript code are accurate, and the code in the PR looks fine to me.

**Comment by nomeata** [comment]
Thanks!

**Pull request merged**